device_backends/include/TriggerDistributor.h

```diff
@@ -104,4 +109,138 @@ namespace ChimeraTK {
         *this, name, numberOfWords, wordOffsetInRegister, flags);
   }
 
+  /********************************************************************************************************************/
+
+  template<typename BackendSpecificDataType>
+  template<typename DistributorType>
+  boost::shared_ptr<DistributorType> TriggerDistributor<BackendSpecificDataType>::getDistributorRecursive(
```

Comment:
Maybe rename into `findDistributor()`?

Reply:
No, it's not internal. The IDE does not find all use cases. I will think of improving the name.

Reply:
I will rename this later, together with the other API changes.
Once the QualifiedAsyncId has been introduces and the TriggerDistributor has been re-named, everything will make more sense.
I don't like "find". I don't have to search for it, I know exactly where it is. And "find" often means that it might not be found, which is not the case because it is created if it does not exist.
Probably "getDistributor()" will be a good name once the code reads
```subDomain->getDistributor(qualifiedAsyncId);```
At the moment it would be
```_distributor->getDistributor(interruptId);```, which is confusing (and it is
```_distributor->getDistributorRecursive(interruptId);``` which is slightly less confusing because the recursive gives a hint why you ask something called distributor for a distributor. It is just a mess at the moment.